### PR TITLE
review chore: Upgrade JDT version

### PR DESCRIPTION
**Pull request by surli**

### CODE REVIEW

**Comment by tdurieux** [comment]
This release is more recent: https://mvnrepository.com/artifact/org.eclipse.tycho/org.eclipse.jdt.core/3.14.0.v20171206-0802

**Comment by surli** [comment]
I'm really not sure about that: the tycho's one has been built on december last year, where the one I proposed has been submitted on Maven this year in April.

Tycho and JDT Core don't follow the same release number.

**Comment by surli** [comment]
And I'm sure in mine Java 10 is supported (see #2054 )

**Comment by tdurieux** [comment]
> Tycho and JDT Core don't follow the same release number.

I'm pretty sure that that the major and minor version number are the same for Tycho and JDT Core

**Comment by surli** [comment]
I just checked locally and apparently Java 10 is not supported in the JDT version you propose.

**Comment by tdurieux** [comment]
eclipse...

**Comment by surli** [comment]
So, OK to merge then ? 😛

**Comment by monperrus** [comment]
LGTM

File: src/test/java/spoon/MavenLauncherTest.java [hunk 1]
```diff
@@ -15,13 +15,22 @@ public void spoonMavenLauncherTest() {
 		// without the tests
 		MavenLauncher launcher = new MavenLauncher("./", MavenLauncher.SOURCE_TYPE.APP_SOURCE);
 
-		assertEquals(15, launcher.getEnvironment().getSourceClasspath().length);
+		// The sourceClasspath seems to change depending on the machine:
+		// I suspect that the config bring in .m2/settings.xml play a role
+		// on my machine (Simon U.) and on spoon3r CI machine we got 2 libraries less
```
Comment (tdurieux) [inline src/test/java/spoon/MavenLauncherTest.java]: #2057 should fix it

**Comment by surli** [comment]
ping @tdurieux @monperrus it's ready

**Comment by monperrus** [comment]
LGTM.

Why does a change in JDT version have an impact on getSourceClasspath?

**Comment by surli** [comment]
> Why does a change in JDT version have an impact on getSourceClasspath?

Because it brings new transitive dependencies I imagine.

**Comment by tdurieux** [comment]
Yes it is the case, I'm still surprise that we double the number of dependencies.

**Comment by surli** [comment]
Just to have an idea, I diff both classpath and here the result:

```
< /Users/urli/.m2/repository/org/eclipse/tycho/org.eclipse.jdt.core/3.13.50.v20171007-0855/org.eclipse.jdt.core-3.13.50.v20171007-0855.jar
---
> /Users/urli/.m2/repository/org/eclipse/jdt/org.eclipse.jdt.core/3.13.102/org.eclipse.jdt.core-3.13.102.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.core.resources/3.12.0/org.eclipse.core.resources-3.12.0.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.core.expressions/3.6.0/org.eclipse.core.expressions-3.6.0.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.core.runtime/3.13.0/org.eclipse.core.runtime-3.13.0.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.osgi/3.12.100/org.eclipse.osgi-3.12.100.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.equinox.common/3.9.0/org.eclipse.equinox.common-3.9.0.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.core.jobs/3.9.3/org.eclipse.core.jobs-3.9.3.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.equinox.registry/3.7.0/org.eclipse.equinox.registry-3.7.0.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.equinox.preferences/3.7.0/org.eclipse.equinox.preferences-3.7.0.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.core.contenttype/3.6.0/org.eclipse.core.contenttype-3.6.0.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.equinox.app/1.3.400/org.eclipse.equinox.app-1.3.400.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.core.filesystem/1.7.0/org.eclipse.core.filesystem-1.7.0.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.text/3.6.100/org.eclipse.text-3.6.100.jar
> /Users/urli/.m2/repository/org/eclipse/platform/org.eclipse.core.commands/3.9.0/org.eclipse.core.commands-3.9.0.jar
```

**Comment by monperrus** [comment]
OK, I'll merge

**Pull request merged**